scipy/ndimage/tests/test_measurements.py

```diff
@@ -698,6 +698,20 @@ def test_median03():
     assert_almost_equal(output, 3.0)
 
 
+def test_median_gh12836_bool():
+    # test boolean addition fix on example from gh-12836
+    a = np.asarray([1, 1], dtype=bool)
+    output = ndimage.median(a, labels=np.ones((2,)), index=[1])
+    assert_array_almost_equal(output, [1.0])
+
+
+def test_median_no_int_overflow():
+    # test integer overflow fix on example from gh-12836
+    a = np.asarray([65, 70], dtype=np.int8)
+    output = ndimage.median(a, labels=np.ones((2,)), index=[1])
+    assert_array_almost_equal(output, [67.5])
```

Comment:
CI is all green and the tests make sense, so looks pretty solid to me.

If the original intention was that when the `labels` and `index` are both provided that the input `dtype` would be preserved in the list of medians, overflow or not, I would suspect that a regression test would already exist for that?

Reply:
The code prior to this PR already always returns floats for integer inputs due to the ` / 2.0` used in the calculation. I think the intention is to give an equivalent result to looping over calls to `numpy.median` for each labelled region.

For example:
```Python
import numpy as np
import scipy.ndimage as ndi
x = np.arange(16, dtype=np.uint8)
assert ndi.median(x) == np.median(x)  # both give 7.5
```